sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/HttpUtil.java

```diff
@@ -99,6 +99,28 @@ public static String post(String url, Map<String, String> headers, String body,
         return result;
     }
 
+    public static HttpResponse postWithResponse(String url, Map<String, String> headers, String body,
+        String contentType) {
+
+        HttpResponse result = null;
+
+        try (CloseableHttpClient client = buildClient()) {
```

Comment:
Do we build a new client for every request? Should we just build it once and reuse the same client?

Reply:
I'm sure we can refactor some of this logic to reuse a client when an application first tries to load up certificates from a vault, but I think that can be a separate PR.